# HG changeset patch
# User Rik <rik@octave.org>
# Date 1404403788 25200
#      Thu Jul 03 09:09:48 2014 -0700
# Node ID f1e21a495e206e3ffb4b0778b870ede5984d80b6
# Parent  b6fdfe22634b4ea18fe157808b5316cbc20c4efb
pareto.m: Use same color for left/right axes and for bar/line.

pareto.m: Copy left-axis color of plotyy to right-axis.  Use colororder from
first axis to set color of line object.

diff --git a/scripts/plot/draw/pareto.m b/scripts/plot/draw/pareto.m
--- a/scripts/plot/draw/pareto.m
+++ b/scripts/plot/draw/pareto.m
@@ -99,17 +99,19 @@ function h = pareto (varargin)
     [ax, hbar, hline] = plotyy (hax, 1 : idx95, y(1:idx95),
                                      1 : length (cdf), 100 * cdf,
                                      @bar, @plot);
   endif
 
   axis (ax(1), [1 - 0.6, idx95 + 0.6, 0, maxcdf]);
   axis (ax(2), [1 - 0.6, idx95 + 0.6, 0, 100]);
   set (ax(2), "ytick", [0, 20, 40, 60, 80, 100],
-              "yticklabel", {"0%", "20%", "40%", "60%", "80%", "100%"});
+              "yticklabel", {"0%", "20%", "40%", "60%", "80%", "100%"},
+              "ycolor", get (ax(1), "ycolor"));
+  set (hline, "color", get (ax(1), "colororder")(1,:));
   set (ax(1:2), "xtick", 1:idx95, "xticklabel", x(1:idx95));
 
   if (nargout > 0)
     h = [hbar; hline];
   endif
 
 endfunction
 
